### PR TITLE
Make fp modules cherry-pickable

### PR DESCRIPTION
**Pull request by epeli**
As discussed in https://github.com/lodash/lodash/issues/1704#issuecomment-170259732
Only a script at this point. Not hooked up in any builds yet.

Easiest way to test this now is just to make it spit out the module wrappers to an existing lodash installation:

```
$ node lib/fp/precompile.js node_modules/lodash/fp
$ node
> var includes = require("lodash/fp/includes");
> includes(1)([2,3,4,1])
true
```

The `includes.placeholder` seems to be undefined. Bug in the `convert` function?

### CODE REVIEW

**Comment by jdalton** [comment]
> The `includes.placeholder` seems to be undefined. Bug in the `convert` function?

Not a bug. The `placeholder` is wired up depending on if it's included in the `lodash.js` build where it would be assigned `lodash`. If not the it's up to the user to specify what they want as a placeholder.

**Comment by jdalton** [comment]
@epeli Awesome! Have you signed [our CLA](http://dojofoundation.org/about/cla)?

**Comment by epeli** [comment]
Now I have.

**Comment by jdalton** [comment]
Rock!

File: lib/fp/precompile.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+`.trim();
+
+var collectionTemplate = name => `
+var convert = require("./convert");
+module.exports = convert(require("../${name}"));
+`.trim();
+
+var passThroughTemplate = name => `module.exports = require("../${name}");`;
+
+var moduleTemplate = name => {
```
Comment (jdalton) [inline lib/fp/precompile.js]: No arrow functions plz. We support Node 0.10.0

File: lib/fp/precompile.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        path.join(target, aliasName + '.js'),
+        moduleTemplate(origName)
+      );
+    }, aliases);
+  }, mapping.aliasMap);
+}
+
+module.exports = precompileFpWrappers;
+
+if (require.main === module) {
```
Comment (jdalton) [inline lib/fp/precompile.js]: No need for the cli branch at the moment.

**Comment by epeli** [comment]
Added support for node 0.10 and removed the cli branch. Rebased and force pushed this PR.

**Comment by jdalton** [comment]
Awesome!

**Pull request merged**